app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/LoadInStorePickupOnGetListPlugin.php

```diff
@@ -7,25 +7,24 @@
 
 namespace Magento\InventoryInStorePickup\Plugin\InventoryApi\SourceRepository;
 
-use Magento\Framework\Api\ExtensionAttributesFactory;
-use Magento\InventoryApi\Api\Data\SourceInterface;
 use Magento\InventoryApi\Api\Data\SourceSearchResultsInterface;
 use Magento\InventoryApi\Api\SourceRepositoryInterface;
-use Magento\InventoryInStorePickupApi\Api\Data\PickupLocationInterface;
+use Magento\InventoryInStorePickup\Model\Source\InitPickupLocationExtensionAttributes;
 
 class LoadInStorePickupOnGetListPlugin
 {
     /**
-     * @var ExtensionAttributesFactory
+     * @var InitPickupLocationExtensionAttributes
      */
-    private $extensionAttributesFactory;
+    private $setExtensionAttributes;
 
     /**
-     * @param ExtensionAttributesFactory $extensionAttributesFactory
+     * @param InitPickupLocationExtensionAttributes $setExtensionAttributes
      */
-    public function __construct(ExtensionAttributesFactory $extensionAttributesFactory)
-    {
-        $this->extensionAttributesFactory = $extensionAttributesFactory;
+    public function __construct(
+        InitPickupLocationExtensionAttributes $setExtensionAttributes
+    ) {
+        $this->setExtensionAttributes = $setExtensionAttributes;
```

Comment:
While this seems to be an excessive extraction, it seems to be the bet solution if we don't figure out to load data in a better way.

Reply:
Well, as a variant, it probably would be nice to extract this logic into a chain of responsibility.